static/js/src/ejsImageGallery.js

```diff
@@ -136,14 +136,17 @@ class ImageGallery {
             const a = document.createElement('a')
             a.href = url
 
+            // eg http://youtu.be/-wtIMTCHWuI ; http://youtu.be/dQw4w9WgXcQ?feature=youtube_gdata_player
             if (a.hostname === 'youtu.be') {
                 videoId = a.pathname.substr(1, a.pathname.length)
                 resolve(videoId)
 
                 return
             }
 
-            videoId = a.search.substr(3, a.search.length)
+            // for generic urls eg:- http://www.youtube.com/watch?v=-wtIMTCHWuI
+            const queryString = new URLSearchParams(a.search)
+            const videoId = queryString.get('v')
```

Comment:
what if there is NO query string param `v`?

example, consider the URL `http://www.youtube.com/watch?foo=bar`.

Revisit fundamentals. Start with the test cases in spreadsheet. Make sure all cases are addressed.